vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/dynamic_certificate.go

```diff
@@ -0,0 +1,228 @@
+package server
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// DynamicCertificateConfig provides a way to have dynamic https configuration for a fixed set of files.
+// You need to remember to call CheckCerts before usage and start the go routine for Run.
+type DynamicCertificateConfig struct {
+	CurrentValue atomic.Value
```

Comment:
doc

Reply:
this type in a config is strange. Don't like it.

Reply:
> this type in a config is strange. Don't like it.

It's the runtime side of the config. It needs this struct, but what you like to call it?

Reply:
`Info`?

Reply:
what is content? What's in there? Untyped fields with unspecific names is bad.